pkgs/development/tools/rust/rust-analyzer/default.nix

```diff
@@ -1,21 +1,27 @@
 { lib, stdenv, fetchFromGitHub, rustPlatform, CoreServices, cmake
 , libiconv
 , useMimalloc ? false
-, doCheck ? true
+# FIXME: Test doesn't pass under rustc 1.52.1 due to different escaping of `'` in string.
+, doCheck ? false
 }:
 
 rustPlatform.buildRustPackage rec {
   pname = "rust-analyzer-unwrapped";
-  version = "2021-06-07";
-  cargoSha256 = "sha256-TyoCu2Q4Tr2EIWxQcjSxASni4dkeEVsfrF5UN7IVxSs=";
+  version = "2021-06-21";
+  cargoSha256 = "sha256-OpfcxBeNwXSD830Sz3o07kgIdXTbZNNVGpaPeCIGGV8=";
 
   src = fetchFromGitHub {
     owner = "rust-analyzer";
     repo = "rust-analyzer";
     rev = version;
-    sha256 = "sha256-f8jdBL42+bU8KKchkW4fF6+kDBjgpoOZyP5yOYsebBk=";
+    sha256 = "sha256-nL5lSvxpOS+fw4iH/Gnl/DI86T9tUtguOy+wLGRkoeY=";
   };
 
+  patches = [
+    # We have rustc 1.52.1 in nixpkgs.
```

Comment:
Can be removed with https://github.com/NixOS/nixpkgs/commit/37ac6d6e760d60d3a99eff4637398c407c68cfe8

Reply:
No. That commit is still in `staging`, but this PR is on `master`. Removing the patch would break the build on `master`.
It seems `staging` is about 3 weeks old but still not merged into `master`. I cannot assume it would be merged soon now.

Reply:
That was a note from me. We can also target staging but I am just merging this.